quickjspp.hpp

```diff
@@ -1868,4 +1888,48 @@ property_proxy<Key>::operator Value() const
 }
 }
 
+inline void Context::enqueueJob(std::function<void()> && job_fcn) {
+    // No need to free the new value. It will be freed automatically when
+    // the function is called.
+    JSValueConst job_val = newValue(std::move(job_fcn)).release();
+    JS_EnqueueJob(ctx, [](JSContext *ctx, int argc, JSValueConst *argv){
+        try
+        {
+            // use Value constructor rather than context.newValue to avoid retrieving the Context object.
+            Value{ctx, argv[0]}.as<std::function<void()>>()();
+        }
+        catch (exception)
+        {
+            return JS_EXCEPTION;
+        }
+        catch (std::exception const & err)
+        {
+            JS_ThrowInternalError(ctx, "%s", err.what());
+            return JS_EXCEPTION;
+        }
+        catch (...)
+        {
+            JS_ThrowInternalError(ctx, "Unknown error");
+            return JS_EXCEPTION;
+        }
+        return JS_UNDEFINED;
+    }, 1, &job_val);
```

Comment:
```suggestion
    auto job_val = newValue(std::move(job_fcn));
    JSValueConst arg = job_val.v;
    int err = JS_EnqueueJob(ctx, [](JSContext *ctx, int argc, JSValueConst *argv){
        try
        {
            assert(argc >= 1);
            js_traits<std::function<void()>>::unwrap(ctx, argv[0])();
        }
        catch (exception)
        {
            return JS_EXCEPTION;
        }
        catch (std::exception const & err)
        {
            JS_ThrowInternalError(ctx, "%s", err.what());
            return JS_EXCEPTION;
        }
        catch (...)
        {
            JS_ThrowInternalError(ctx, "Unknown error");
            return JS_EXCEPTION;
        }
        return JS_UNDEFINED;
    }, 1, &arg);
    if(err < 0)
        throw exception{ctx};
```
job_fcn can also be optimized to not be wrapped in std::function but whatever.

Reply:
you mean using a templated `Functor`?

Reply:
Yes, something like this
```cpp
    template <typename Function>
    void enqueueJob(Function&& f)
    {
         JSValue job_val = js_traits<std::function<void()>>::wrap(ctx, std::forward<Function>(f));
...
JS_FreeValue(job_val);
```